prompt/prompt-model/src/commonMain/kotlin/ai/koog/prompt/message/Message.kt

```diff
@@ -155,6 +160,37 @@ public sealed interface Message {
         override fun copy(updatedMetaInfo: ResponseMetaInfo): Assistant = this.copy(metaInfo = updatedMetaInfo)
     }
 
+    /**
+     * Represents a reasoning message exchanged in a chat system, encapsulating the content,
+     * role, and associated metadata, with an optional reference to the original thinking process.
+     *
+     * @property id An optional identifier for the reasoning process.
+     * @property encrypted The encrypted content of the reasoning message.
+     * @property content The content of the message as a string.
+     * @property role The [Role] of the message, indicating its source or function in the chat (e.g., assistant, user).
+     *                Defaults to [Role.Assistant].
+     * @property metaInfo Metadata associated with the reasoning, captured as a [ResponseMetaInfo] instance.
+     *                    Provides details such as token usage and timestamps.
+     */
+    @Serializable
+    public data class Reasoning(
+        public val id: String? = null,
+        public val encrypted: String? = null,
+        override val parts: List<ContentPart>,
```

Comment:
Let's limit it to the `List<ContentPart.Text>` since only text content is supported here